File: highestNoPossible/highestNoPossible.js
function descendingOrder(n) {
  /* takes any non-negative integer as an argument and return it with 
    its digits in descending order. Essentially, rearrange the digits to 
    create the highest possible number.*/
  let numS = n
    .toString()
    .split('')
    .sort((a, b) => b - a)
    .join('');
  return Number(numS);
}

// best soln:
function descendingOrder(n) {
  return parseInt(String(n).split('').sort().reverse().join(''));
}

tests;
// const { assert } = require("chai")

// describe("Basic tests", () => {
//   it("Testing for fixed tests", () => {
//     assert.strictEqual(descendingOrder(0), 0)
//     assert.strictEqual(descendingOrder(1), 1)
//     assert.strictEqual(descendingOrder(111), 111)
//     assert.strictEqual(descendingOrder(15), 51)
//     assert.strictEqual(descendingOrder(1021), 2110)
//     assert.strictEqual(descendingOrder(123456789), 987654321)
//     })
//   })

File: highestNoPossible/highestNoPossible.py
def descending_order(num):
    ''' takes any non-negative integer as an argument and return it with 
    its digits in descending order. Essentially, rearrange the digits to 
    create the highest possible number. '''
    l = list(str(num))
    sorted_l = sorted(l, key=lambda x: int(x), reverse=True)
    return int(''.join(sorted_l))

# best soln:


def Descending_Order(num):
    return int("".join(sorted(str(num), reverse=True)))

# tests
# try:
#     from solution import Descending_Order as descending_order
# except ImportError:
#     from solution import descending_order

# @test.describe("Fixed Tests")
# def fixed_tests():
#     @test.it('Basic Test Cases')
#     def basic_test_cases():
#         test.assert_equals(descending_order(0), 0)
#         test.assert_equals(descending_order(15), 51)
    test.assert_equals(descending_order(123456789), 987654321)
